components/omnibox/browser/brave_omnibox_edit_model.cc

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/omnibox/browser/brave_omnibox_edit_model.h"
+
+#include "brave/components/commander/common/commander_frontend_delegate.h"
+#include "brave/components/commander/common/commander_url.h"
+#include "components/omnibox/browser/omnibox_edit_model.h"
+
+BraveOmniboxEditModel::~BraveOmniboxEditModel() = default;
+
+void BraveOmniboxEditModel::OpenMatch(
+    AutocompleteMatch match,
+    WindowOpenDisposition disposition,
+    const GURL& alternate_nav_url,
+    const std::u16string& pasted_text,
+    size_t index,
+    base::TimeTicks match_selection_timestamp) {
+  uint32_t command_index;
+  uint32_t result_set_id;
+  if (commander::TryParseCommandURL(match.destination_url, &command_index,
+                                    &result_set_id)) {
+    commander::CommanderFrontendDelegate::Get()->SelectCommand(command_index,
+                                                               result_set_id);
+    return;
+  }
```

Comment:
I'm not 100% sure but, if we have something like `bool OmniboxEditModel::TryHandleCommand()` here and delegate things to `chrome` layer, it might help you to avoid copying CommandModel for components/ .

Reply:
Unfortunately I think I'll need some kind of intermediate representation anyway, as I need to give something to the AutocompleteProvider.

Maybe I could turn the `CommandItemViewModels` into `AutoCompleteMatches` in the `CommandCentre`? 

At the moment, I'm almost wondering whether it's worth using the Chromium `CommanderBackend` and `CommanderFrontend` classes at all, because I feel like they aren't buying me much - I'm going to have a think about it overnight :smile:

Reply:
Yeah, if you think they're not that useful, having our own representation and control over them could be better, when we think of rebase or extension of this feature!

Reply:
I think it was a good call - Chromium isn't sure if they're going to remove the feature:
https://bugs.chromium.org/p/chromium/issues/detail?id=1014639#c100